src/dimension/Dimension.js

```diff
@@ -0,0 +1,488 @@
+        this._isMultiDim = this._expression.length > 1
+        this._columns    = _mapColumnsToNameAndType(crossfilter.getColumns())
+        // this the collection of columns, expressed as strings
+        // can also cast
+        this._initDimArray(this._expression, crossfilter)
+        this._initDimContainsArray(crossfilter)
+        this._initGroup()
+    }
+    _initDimArray(expression) {
+        this._dimArray = expression.map((field) => {
+            let indexOfColumn   = _findIndexOfColumn(this._columns, field),
+                isDate          = indexOfColumn > -1 && _isDateField(this._columns[indexOfColumn])
+            if (isDate) {
+                field = "CAST(" + field + " AS TIMESTAMP(0))"
+            }
+            return field
+        })
+        this._dimensionExpression = this._dimArray.includes(null) ? null : this._dimArray.join(", ")
+    }
+    _initDimContainsArray(crossfilter) {
+        const { _dimArray }      = this,
+            { _columnTypeMap, _compoundColumnMap }  = crossfilter
+
+        _dimArray.forEach((dim, i) => {
+            if (dim in _columnTypeMap) {
+                this._dimContainsArray[i] = _columnTypeMap[dim].is_array
+            }
+            else if (dim in _compoundColumnMap) {
+                this._dimContainsArray[i] = _columnTypeMap[_compoundColumnMap[dim]].is_array
+            }
+            else {
+                this._dimContainsArray[i] = false
+            }
+        })
+    }
+    _initGroup(dataConnector) {
+        // tbd
+    }
+    // todo - maybe this or some other technique...?
+    _addPublicAPI(crossfilter) {
+        this.writeTopBottomQuery    = writeTopBottomQuery
+        // this.writeFilter = (queryBinParams) => writeGroupFilter(queryBinParams, this)
+        this.top                    = (k, offset, renderSpec, callback) => top(this, k, offset, renderSpec, callback)
+        this.writeTopQuery          = (k, offset, isRender) => writeTopQuery(this, k, offset, isRender)
+        this.writeTopBottomQuery    = (k, offset, ascDescExpr, isRender) => writeTopBottomQuery(this, k, ascDescExpr, isRender)
+        this.bottom                 = (k, offset, renderSpec, callback) => bottom(this, k, offset, renderSpec, callback)
+        this.writeBottomQuery       = (k, offset, isRender) => writeBottomQuery(this, k, offset, isRender)
+        // todo - temporary hack to support backwards compatibility
+        this.remove = this.dispose  = () => this.getCrossfilter().removeDimension(this)
+        this.value                  = () => this._dimArray
+        this.getCrossfilterId       = () => crossfilter.getId()
+        this.getProjectOn           = () => this._projectExpressions
+        this.setDrillDownFilter     = (value) => {
+            this._drillDownFilter = value
+            return this
+        }
+        this.getTable               = crossfilter.getTable
+        // testing haxx
```

Comment:
be more specific about what the hack is and under what circumstances it can be removed